src/translator/service.cpp

```diff
@@ -9,127 +9,61 @@ namespace marian {
 namespace bergamot {
 
 Service::Service(Ptr<Options> options)
-    : requestId_(0), numWorkers_(options->get<int>("cpu-threads")),
-      vocabs_(std::move(loadVocabularies(options))),
-      text_processor_(vocabs_, options), batcher_(options)
-#ifdef WITH_PTHREADS
-      ,
-      pcqueue_(2 * options->get<int>("cpu-threads"))
-#endif // WITH_PTHREADS
-{
+    : ServiceBase(options), numWorkers_(options->get<int>("cpu-threads")),
+      pcqueue_(numWorkers_) {
+  if (numWorkers_ <= 0) {
+    ABORT("Fatal: numWorkers should be greater than 1");
```

Comment:
`ABORT` message and `if` condition doesn't imply same here. A typo?

size_t is anyway unsigned. Therefore, it can never be less than 0.

Did you mean this:

```
if (numWorkers_ == 0) {
  ABORT("Fatal: numWorkers should be greater than 0");
}

```

Reply:
Covered in https://github.com/browsermt/bergamot-translator/pull/44/commits/7c14b737a4202404d1f971960f7341cd0892bd0b.